### PR TITLE
docs: update page on available models

### PR DESCRIPTION
**Pull request by LMMilliken**
This pr updates the section on choosing a model. The different tasks have been moved out of tabs and given their own headers to make it easier to reference them from notebooks or other pages.

---

- [x] This PR references an open issue
- [ ] I have added a line about this change to CHANGELOG

### CODE REVIEW

**Review by bwanglzu — APPROVED** [review]
LGTM!

File: docs/walkthrough/choose-backbone.md [hunk 1]
```diff
@@ -2,48 +2,30 @@
 # Backbone Model
 
 Finetuner provides several widely used backbone models,
-including `resnet`, `efficientnet`, `clip` and `bert`.
-Thereby, for most of them, Finetuner provides multiple variants, e.g., the common `resnet50 ` and the more complex `resnet152` model.
+including `resnet`, `efficientnet`, `clip`, `bert` and `pointnet`.
+Each of these models have their own variants and are designed to produce embeddings for a specific modality.
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:6]:
```suggestion
Each model has its own variants and is designed to produce embeddings for a specific modality.
```

**Review by azayz — APPROVED** [review]
LGTM

File: docs/walkthrough/choose-backbone.md [hunk 2]
```diff
@@ -54,8 +36,26 @@ to get a list of supported models:
 │ sbert-base-en │ text-to-text │        768 │  transformer │                  Pretrained BERT, fine-tuned on MS Marco │
 └───────────────┴──────────────┴────────────┴──────────────┴──────────────────────────────────────────────────────────┘
 ```
-````
-````{tab} image-to-image
+
+To get a list of supported models during run-time, you can call:
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:40]:
```suggestion
To get a list of supported models at run-time, you can call:
```

File: docs/walkthrough/choose-backbone.md [hunk 3]
```diff
@@ -67,8 +67,27 @@ to get a list of supported models:
 │        resnet-base │ image-to-image │       2048 │          cnn │        ResNet50 pre-trained on ImageNet │
 └────────────────────┴────────────────┴────────────┴──────────────┴─────────────────────────────────────────┘
 ```
-````
-````{tab} text-to-image
+
+To get a list of supported models during run-time, you can call:
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:71]:
```suggestion
To get a list of supported models at run-time, you can call:
```

File: docs/walkthrough/choose-backbone.md [hunk 4]
```diff
@@ -67,8 +67,27 @@ to get a list of supported models:
...
+
+```python
+import finetuner
+
+finetuner.describe_models(task='image-to-image')
+```
+
+## Text-to-Image Models
+
+CLIP models are used to text-to-image search tasks, and we support two different types.
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:81]:
```suggestion
CLIP models are used for text-to-image search tasks. We support two different types of CLIP models:
```

File: docs/walkthrough/choose-backbone.md [hunk 5]
```diff
@@ -67,8 +67,27 @@ to get a list of supported models:
...
+- `clip-base-en` and `clip-large-en` are both CLIP models trained only on English text,
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:82]:
```suggestion
- English only models `clip-base-en` and `clip-large-en`,
```

File: docs/walkthrough/choose-backbone.md [hunk 6]
```diff
@@ -67,8 +67,27 @@ to get a list of supported models:
...
+  with `clip-large-en` requiring more computational resources than `clip-base-en`.
+- `clip-base-multi` is a multilingual model that has been trained on the [laion5b](https://laion.ai/blog/laion-5b/)
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:84]:
```suggestion
- A multilingual model `clip-base-multi`, trained on the [laion5b](https://laion.ai/blog/laion-5b/)
```

File: docs/walkthrough/choose-backbone.md [hunk 7]
```diff
@@ -67,8 +67,27 @@ to get a list of supported models:
...
+  dataset, which contains training from 100+ different langauges. 
+  You should use this model when encoding non-English text.
+
+Similarly to the models used for text-to-text tasks, these models are transformer-based, and are loaded from the huggingface
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:88]:
```suggestion
Like the models used for text-to-text tasks, these models are transformer-based, and are loaded from the huggingface
```

File: docs/walkthrough/choose-backbone.md [hunk 8]
```diff
@@ -80,8 +99,21 @@ to get a list of supported models:
...
+
+```python
+import finetuner
+
+finetuner.describe_models(task='text-to-image')
+```
+
+## Mesh-to-Mesh Models
+
+Currently, we only support one model for mesh-to-mesh search, `pointnet-base`, which we derived from the popular
```
Comment (scott-martens) [inline docs/walkthrough/choose-backbone.md:113]:
```suggestion
Currently, we only support one model for mesh-to-mesh search, `pointnet-base`, which we derived from the popular 
```

**Review by guenthermi — CHANGES_REQUESTED** [review]
Added some comment. 
Some things which I think are missing are:
- some information about the `display_name` and the `descriptor`
- a small noticed that the names are used in the fit function, may in one of the first sentences.
- Maybe ask Bo about this, but we could also describe how to build the pre-trained models. Though, how to use them to encode something should be explained in a different section
- You can also already add descriptions for the models used in the synthesis task. We will soon add multi-lingual models for synthesis. Then this can be extended.

File: docs/walkthrough/choose-backbone.md (hunk 8 above)
Comment (LMMilliken) [reply docs/walkthrough/choose-backbone.md:113]: Did you mean to suggest a change here?

File: docs/walkthrough/choose-backbone.md [hunk 9]
```diff
@@ -2,48 +2,32 @@
 # Backbone Model
 
 Finetuner provides several widely used backbone models,
-including `resnet`, `efficientnet`, `clip` and `bert`.
-Thereby, for most of them, Finetuner provides multiple variants, e.g., the common `resnet50 ` and the more complex `resnet152` model.
+including `resnet`, `efficientnet`, `clip`, `bert` and `pointnet`.
+Each model has its own variants and are designed to produce embeddings for a specific modality.
```
Comment (gmastrapas) [inline docs/walkthrough/choose-backbone.md:6]:
```suggestion
Each model has its own variants and is designed to produce embeddings for a specific modality.
```

File: docs/walkthrough/choose-backbone.md [hunk 10]
```diff
@@ -2,48 +2,32 @@
...
+- `bert-base-en` is a raw pre-trained model, and should be used if you want to train something from 
+  scratch or train a model on a task where a notion of similarity you want to finetune on is very different
+  from what's usually considered as similar.
+- `sbert-base-en` has been fine-tuned once by [sentence-transformers](https://www.sbert.net/) on the 
+  [MS MARCO](https://microsoft.github.io/msmarco/) dataset on top of BERT.
+  It is designed for matching web search queries to short text passages and 
+  is a suitable backbone for traditional text-to-text search tasks.
 
-to get a list of supported models:
+These transformer-based models are loaded from the huggingface
```
Comment (gmastrapas) [inline docs/walkthrough/choose-backbone.md:25]:
```suggestion
These transformer-based models are loaded using the huggingface
```

File: docs/walkthrough/choose-backbone.md [hunk 11]
```diff
@@ -54,8 +38,26 @@ to get a list of supported models:
...
+
+finetuner.describe_models(task='text-to-text')
+```
+
+## Image-to-Image Models
+
+For image-to-image tasks, we support two different types of models, `efficientnet` and `resnet`,
+both of which having `base` and `large` variations.
+- ResNets are suitable for image-to-image search tasks with high performance requirements, where resnet152 is bigger and
+  requires higher computational resources than resnet50.
```
Comment (gmastrapas) [inline docs/walkthrough/choose-backbone.md:55]:
```suggestion
  requires higher computational resources than resnet50 (base variant).
```

File: docs/walkthrough/choose-backbone.md [hunk 12]
```diff
@@ -67,8 +69,27 @@ to get a list of supported models:
...
+## Text-to-Image Models
+
+CLIP models are used for text-to-image search tasks. We support two different types of CLIP models:
+- English-only models `clip-base-en` and `clip-large-en`,
+  with `clip-large-en` requiring more computational resources than `clip-base-en`.
+- A multilingual model `clip-base-multi`, trained on the [laion5b](https://laion.ai/blog/laion-5b/)
+  dataset, which contains data from 100+ different langauges. 
+  You should use this model when encoding non-English text.
+
+Like the models used for text-to-text tasks, these models are transformer-based, and are loaded from the huggingface
```
Comment (gmastrapas) [inline docs/walkthrough/choose-backbone.md:90]:
```suggestion
Like the models used for text-to-text tasks, these models are transformer-based, and are loaded using the huggingface
```

File: docs/walkthrough/choose-backbone.md [hunk 13]
```diff
@@ -90,18 +124,91 @@ to get a list of supported models:
...
+## Model Options
+
+Once you have chosen a model you can provide a dictionary of options to adjust the behaviour of the model to the
+`model_options` parameter of the {meth}`finetuner.fit` function.
+You can see a list of additional options that you can provide for each model using the
+{meth}`finetuner.list_model_options` function:
+
+![get_model_options](../imgs/get_model_options.png)
+
+While each model has their own set of options, the `sbert_base_en` model, as shown in the example above, has the four most common options.
```
Comment (gmastrapas) [inline docs/walkthrough/choose-backbone.md:145]:
```suggestion
While each model has their own set of options, the `sbert-base-en` model, as shown in the example above, has the four most common options.
```

**Review by gmastrapas — APPROVED** [review]

**Pull request merged**